pluginops/prs.go

```diff
@@ -0,0 +1,249 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/go-github/v32/github"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/slices"
+)
+
+func init() {
+	rootCmd.AddCommand(prsCmd)
+
+	prsCmd.AddCommand(
+		prsListCmd,
+		prsMergeCmd,
+	)
+}
+
+var prsCmd = &cobra.Command{
+	Use:   "prs",
+	Short: "Manage PRs.",
+}
+
+var prsListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List your plugin PRs.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := getGitHubClient()
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		prs, err := getPluginPRs(ctx, client)
+		if err != nil {
+			return err
+		}
+
+		slices.SortFunc(prs, func(a, b *prInfo) bool { return a.Repository < b.Repository })
+
+		for _, pr := range prs {
+			out := fmt.Sprintf("%s: %s", pr.Repository, pr.GetTitle())
+
+			if pr.readyToBeMerged() {
+				out += Green("\u2713")
+			} else {
+				out += Red("\u2715")
+			}
+			fmt.Print(out + "\n")
+		}
+
+		return nil
+	},
+}
+
+var prsMergeCmd = &cobra.Command{
+	Use:   "merge",
+	Short: "Merge plugin PRs that are ready.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := getGitHubClient()
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		prs, err := getPluginPRs(ctx, client)
+		if err != nil {
+			return err
+		}
+
+		if len(prs) == 0 {
+			fmt.Println("No PRs to merge.")
+			return nil
+		}
+
+		slices.SortFunc(prs, func(a, b *prInfo) bool { return a.Repository < b.Repository })
+
+		for _, pr := range prs {
+			if !pr.readyToBeMerged() {
+				log.WithFields(log.Fields{"repo": pr.Repository, "title": pr.GetTitle()}).Debug("PR not ready to be merged")
+				continue
+			}
+
+			ok, err := confirmPrompt(fmt.Sprintf("Merge %s/%s", pr.Repository, pr.GetTitle()))
+			if err != nil {
+				return err
+			}
+			if !ok {
+				continue
+			}
+
+			err = mergePR(context.Background(), client, pr)
+			if err != nil {
+				return err
+			}
+
+			fmt.Printf("Merged %s\n\n", pr.GetHTMLURL())
+		}
+
+		return nil
+	},
+}
+
+type prInfo struct {
+	Repository string
+	*github.PullRequest
+	reviews []*github.PullRequestReview
+}
+
+func getPluginPRs(ctx context.Context, client *github.Client) ([]*prInfo, error) {
+	user, _, err := client.Users.Get(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+
+	query := fmt.Sprintf("is:pr is:open author:%v org:%v", user.GetLogin(), org)
+
+	issues, _, err := client.Search.Issues(ctx, query, &github.SearchOptions{}) // TODO: handle pagination
+	if err != nil {
+		return nil, err
+	}
+
+	var prs []*prInfo
+
+	var wg sync.WaitGroup
+	for _, issue := range issues.Issues {
+		if issue.IsPullRequest() {
```

Comment:
To avoid indentation

```suggestion
		if !issue.IsPullRequest() {
			continue
		}
```

Reply:
How could this happen though? We are using `is:pr` in the query

Reply:
It's an extra check that should not be nessary. #defensiveCode